library/core/class.pluginmanager.php

```diff
@@ -931,7 +931,9 @@ public function scanPluginFile($PluginFile, $VariableName = null) {
 
         }
         unset($Lines);
-        if ($PluginInfoString != '') {
+        if (empty($PluginInfoString)) {
+            return null;
+        } else {
```

Comment:
These changes prevent an issue where plugins using the correct addon.json would fail the forwards compatibility test.